compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/draganddrop/WebDragAndDropManager.kt

```diff
@@ -0,0 +1,187 @@
+package androidx.compose.ui.draganddrop
+
+import androidx.compose.ui.events.EventTargetListener
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Canvas
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.drawscope.CanvasDrawScope
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.platform.PlatformDragAndDropManager
+import androidx.compose.ui.platform.PlatformDragAndDropSource
+import androidx.compose.ui.scene.ComposeSceneDragAndDropNode
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.LayoutDirection
+import kotlin.math.roundToInt
+import kotlinx.browser.document
+import kotlinx.browser.window
+import org.khronos.webgl.Uint8ClampedArray
+import org.w3c.dom.CanvasRenderingContext2D
+import org.w3c.dom.DragEvent
+import org.w3c.dom.HTMLCanvasElement
+import org.w3c.dom.ImageData
+import androidx.compose.ui.graphics.asByteArray
+import org.w3c.dom.HTMLElement
+
+internal abstract class WebDragAndDropManager(eventListener: EventTargetListener, private val density: Density) :
+    PlatformDragAndDropManager {
+    override val isRequestDragAndDropTransferRequired: Boolean
+        get() = false
+
+    abstract val rootDragAndDropNode: ComposeSceneDragAndDropNode
+
+    private val startTransferScope = object : PlatformDragAndDropSource.StartTransferScope {
+        /**
+         * Context for an ongoing drag session initiated from Compose.
+         */
+        var dragSessionContext: DragSessionContext? = null
+
+
+        override fun startDragAndDropTransfer(
+            transferData: DragAndDropTransferData,
+            decorationSize: Size,
+            drawDragDecoration: DrawScope.() -> Unit
+        ): Boolean {
+            dragSessionContext = DragSessionContext()
+
+            val imageBitmap = ImageBitmap(
+                width = decorationSize.width.roundToInt(),
+                height = decorationSize.height.roundToInt()
+            )
+
+            // This results in blurry text for some reason.
```

Comment:
As far as I remember @m-sasha had similar problem on Desktop

Reply:
I'll remove this comment since in our case it was not a problem after some fixes introduced